packages/core/docs/advanced/caching.md

```diff
@@ -1,99 +1,133 @@
+3. Page is served to the user.
 
-const cache = useCache(); // Inside setup
-cache.addTags([{ prefix: 'P', value: '123' }]) // adds tags
-cache.setTags((currentTags) => [...currentTags, { prefix: 'P', value: '123' }]) // adds tags, also you can skip previous one
-cache.getTags(); // returns current tags
-```
+## Using tags
 
-## Nuxt compatibility
-Add it to **modules** not **buildModules** - it is very important.
+Tags should be registered in Vue components or composables. During Server Side Rendering, tags registered in the current route are associated with the rendered page.
 
-```js
-['@vue-storefront/cache/nuxt', {
-  server: {
-    invalidateEndpoint: '/cache-invalidate',
-    invalidators: ['./invalidator'],
-    driver: ['./exampleCache', { test: 1 }]
+To add tags, use `useCache` composable from `@vue-storefront/cache` package.
+
+```javascript
+// pages/Category.vue
+import { useCache, CacheTagPrefix } from '@vue-storefront/cache';
+
+export default {
+  setup() {
+    const { addTags } = useCache();
+
+    addTags([
+      { prefix: CacheTagPrefix.View, value: 'category' },
+      { prefix: CacheTagPrefix.Category, value: id },
+      // or
+      { prefix: 'V', value: 'category' },
+      { prefix: 'C', value: id },
+    ]);
   }
-}]
+};
 ```
 
-## Tagging from integration composables
+Tags should be added for all dynamic parts of the page. For example on the category page, tags should be added for:
+  * current category,
+  * all visible subcategories,
+  * all visible products.
 
-```ts
-const factoryParams = {
-  getCacheTags: (response): CacheTags[] => {
-    return [
-      prefix: 'P', value: '123' },
-      prefix: 'P', value: '456' },
-      prefix: 'P', value: '789' }
-    ]
-  }
+**But why?**
+
+When at least one tag associated with the given page is [invalidated](#invalidating-cache), the whole page is removed from the cache. For example, if one of the products is modified or disabled, we should invalidate cache for pages where it's visible:
+* Product page for this product.
+* Other product pages, where this product is listed (upsell or cross-sell).
+* Homepage, if the product is displayed in the carousel or listed as a popular item.
+* Category page, where this product is listed.
+* Search page, where this product is part of the results.
+
+Additionally, all modifiers changing what is displayed on the page, such as pagination, filtering, and sorting options should be added as URL queries (for example `?sort=price-up&size=36&page=3`). This will cause different modifier combinations to be treated as different routes, and thus, caching them separately.
+
+::: warning
+Don't use tags on pages, components, or composables specific to the current user, such as user profile pages or cart components.
+:::
+
+## Invalidating cache
+
+As mentioned in [Configuration](#configuration) section, `@vue-storefront/cache` module provides option to create invalidation endpoint.
```

Comment:
some explaination about the invalidation could be useful, perhaps moving parts about invalidation from above here will make sense as well.

When users are reading docs they're usually looking only at specific paragraphs with solutions to their problems instead of reading them as a whole.